java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java

```diff
@@ -230,24 +230,17 @@ private void handleDisconnect() {
             return;
         }
 
-        resultTable.sealTable(() -> {
-            completed = true;
-            signalCompletion();
-        }, () -> {
-            exceptionWhileCompleting = new Exception();
-            signalCompletion();
-        });
+        completed = true;
+        signalCompletion();
         cleanup();
     }
 
-    private void signalCompletion() {
+    private synchronized void signalCompletion() {
```

Comment:
Something is fishy in this class. `exceptionWhileCompleting` exists, but is never assigned to non-null.